Description: Set folderview to point at Desktop folder if running from a live CD.
Author: Jonathan Riddell <jriddell@ubuntu.com>
Origin: vendor
Forwaded: not-needed
Bug-Ubuntu: https://launchpad.net/bugs/613834
Index: kde-baseapps-4.10.3/plasma/applets/folderview/folderview.cpp
===================================================================
--- kde-baseapps-4.10.3.orig/plasma/applets/folderview/folderview.cpp	2013-05-13 10:17:46.000000000 +0000
+++ kde-baseapps-4.10.3/plasma/applets/folderview/folderview.cpp	2013-05-13 10:17:46.000000000 +0000
@@ -27,6 +27,7 @@
 #include <QGraphicsSceneDragDropEvent>
 #include <QImageReader>
 #include <QItemSelectionModel>
+#include <QFile>
 
 #include <KAction>
 #include <KAuthorized>
@@ -430,6 +431,10 @@
                 path = QString("desktop:/");
             }
         }
+
+        if (QFile::exists("/usr/share/applications/kde4/oem-config-prepare-kde.desktop") || QFile::exists("/usr/share/applications/kde4/ubiquity-kdeui.desktop")) {
+            path = QString("desktop:/");
+        }
         setUrl(cg.readEntry("url", KUrl(path)));
     }
 
